### PR TITLE
[DBC-124] [DBC-125] Implement business logic in Token details screen

### PR DESCRIPTION
**Pull request by quocnd**
- [Get a transaction history list for the given Token](https://mvlchain.atlassian.net/browse/DBC-124)
- [Get token balance](https://mvlchain.atlassian.net/browse/DBC-125)

### CODE REVIEW

File: app/src/main/java/io/mvlchain/tada/wallet/app/ui/tokendetails/TransactionHistoryAdapter.kt [hunk 1]
```diff
@@ -17,69 +17,83 @@ package io.mvlchain.tada.wallet.app.ui.tokendetails
...
-                ),
-                parent, false
-            )
-        )
+        return TransactionHistoryViewHolder.create(parent)
     }
 
     override fun onBindViewHolder(holder: TransactionHistoryViewHolder, position: Int) {
-        holder.bind(listener, historyList[position])
+        holder.bind(listener, getItem(position)!!)
```
Comment (cutehackers) [inline app/src/main/java/io/mvlchain/tada/wallet/app/ui/tokendetails/TransactionHistoryAdapter.kt:61]: We do not use !! operator that's mvl's rule. Please avoid using !! operator
like -> 
`getItem(position)?.also { holder.bind(listener, it) }`

File: app/src/main/java/io/mvlchain/tada/wallet/app/ui/tokendetails/TransactionLoadingStateAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.paging.LoadState
+import androidx.paging.LoadStateAdapter
+import androidx.recyclerview.widget.RecyclerView
+import io.mvlchain.tada.wallet.databinding.ItemNetworkStateBinding
+
+class TransactionLoadingStateAdapter(
```
Comment (cutehackers) [inline app/src/main/java/io/mvlchain/tada/wallet/app/ui/tokendetails/TransactionLoadingStateAdapter.kt:26]: TransactionLoadingStateAdapter -> TransactionLoadStateAdapter ?

File: app/src/main/res/navigation/main_graph.xml [hunk 3]
```diff
@@ -102,9 +105,13 @@
         android:name="io.mvlchain.tada.wallet.app.ui.tokendetails.TokenDetailsFragment"
         android:label="TokenDetailsFragment"
         tools:layout="@layout/fragment_token_details">
+
         <argument
-            android:name="ticker"
+            android:name="wallet_id"
```
Comment (cutehackers) [inline app/src/main/res/navigation/main_graph.xml:110]: should be camel case format -> walletId

File: app/src/test/java/io/mvlchain/tada/wallet/app/util/DateTimeFormatterTest.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+import junit.framework.TestCase
+import org.junit.Test
+import java.util.Date
+import java.util.Calendar
+
+class DateTimeFormatterTest {
+
+    @Test
+    fun testFormat() {
```
Comment (cutehackers) [inline app/src/test/java/io/mvlchain/tada/wallet/app/util/DateTimeFormatterTest.kt:26]: It's good to have test code.  wonderful

File: shared/src/main/java/io/mvlchain/tada/wallet/shared/data/api/response/SimpleWalletResponse.kt [hunk 5]
```diff
@@ -20,9 +20,12 @@ import io.mvlchain.tada.wallet.shared.data.db.entity.AssetEntity
 import io.mvlchain.tada.wallet.shared.data.db.entity.BalanceEntity
 import io.mvlchain.tada.wallet.shared.data.db.entity.SimpleWalletEntity
 
+/**
+ * @property network BlockChainNetwork
+ */
 data class SimpleWalletResponse(
     @SerializedName("id")
-    val id: String,
+    var id: String,
```
Comment (cutehackers) [inline shared/src/main/java/io/mvlchain/tada/wallet/shared/data/api/response/SimpleWalletResponse.kt:28]: any reason why you change from val to var?

**Review by cutehackers — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/mvlchain/tada/wallet/app/ui/tokendetails/TransactionHistoryAdapter.kt (hunk 1 above)
Comment (quocnd) [reply app/src/main/java/io/mvlchain/tada/wallet/app/ui/tokendetails/TransactionHistoryAdapter.kt:61]: Thank for pointing it out, will change it.

File: shared/src/main/java/io/mvlchain/tada/wallet/shared/data/api/response/SimpleWalletResponse.kt (hunk 5 above)
Comment (quocnd) [reply shared/src/main/java/io/mvlchain/tada/wallet/shared/data/api/response/SimpleWalletResponse.kt:28]: I planned to generate multiple wallets to test in `FakeWalletApi`, that need to modify `Id`. But it's no need at this moment, so I will reverse it.
Note: Some other places e.g: `id` in `TransactionHistoryResponse` will be changed to `val` once we integrate with backend API

Comment (cutehackers) [reply shared/src/main/java/io/mvlchain/tada/wallet/shared/data/api/response/SimpleWalletResponse.kt]: in that case, you can copy() method of data class so you can keep SimpleWalletResponse instance as it is.

Comment (cutehackers) [reply shared/src/main/java/io/mvlchain/tada/wallet/shared/data/api/response/SimpleWalletResponse.kt]:
```
val response: SimpleWalletResponse
ex) val newResponse = response.copy(id = "new_id")
```
newResponse is going to be a new instance with new value "new_id"

**Review by cutehackers — APPROVED** [review]

**Pull request merged**